src/asciidoctor-core-api.js

```diff
@@ -365,8 +370,28 @@ AbstractNode.$$proto.getAttributes = function () {
  * @memberof AbstractNode
  */
 AbstractNode.$$proto.getAttribute = function (name, defaultValue, inherit) {
-  var value = this.$attr(name, defaultValue, inherit);
```

Comment:
In fact, things are a little bit more complex since we may insert an attribute both using the setAttribute method. But also by using hash_put. And that later has specific handling for non-string keys.

Using asciidcotor.js from the master branch, and given an attribute list containing only one positional argument:

$Hash {
  '$$smap': { style: 'positional1' },
  '$$map': { '3': { key: 1, key_hash: 3, value: 'positional1' } },
  '$$keys': [ { key: 1, key_hash: 3, value: 'positional1' }, 'style' ],
  '$$none': undefined,
  '$$proc': undefined }
> block.getAttribute(1)
undefined
> block.getAttribute('1')
undefined
> Opal.hash_get(block.attributes, 1)
'positional1'
> Opal.hash_get(block.attributes, '1')
undefined

As you can see, `getAttribute` returns undefined both for the number key, but also for the string representation of that key since the Opal hash stores objects keys using its own internal format.

Indeed, we could forward the special case handling to `$attr`. As I said before, I didn't know to which extent this would impact the rest of the code. But if you're OK with that, no problem in doing it.

Reply:
I think it really is time to step back and make a more decisive statement. Attribute names must be strings. We shouldn't be trying to expose these numeric keys at all. A separate method should provide read-only access to them. getAttributes() should filter these numeric keys out and getAttribute() should not be able to see them.

Reply:
@mojavelinux 
Concretely what is the use case for positional arguments?
The whole discussion started because the attribute list contains numeric keys introduced from https://github.com/asciidoctor/asciidoctor/blob/9aefdeed07b2a72b7518c80b1e67239b433f6018/lib/asciidoctor/attribute_list.rb#L183

```
      # QUESTION should we always assign the positional key?
      @attributes[index + 1] = resolved_name
      # QUESTION should we assign the resolved name as an attribute?
      #@attributes[resolved_name] = nil
```

According to the comments there, it is not even obvious positional arguments should be stored in the first place into the attribute list.

Reply:
...which is what I've been trying to say ;)

Reply:
OK :)
Let's say it _should_ be removed from the core, but in the meantime, I will filter out non-string keys from the getAttributes() result.